### PR TITLE
Update to TRIQS 3 and Python 3

### PR DESCRIPTION
**Pull request by Wentzell**
This PR will update pomerol2triqs to Python 3 and the latest TRIQS release (3.0.0).

### CODE REVIEW

**Comment by krivenko** [comment]
Thank you so much, Nils!

I must try to build and run this on my machine before I merge.

The only question I have right now is whether it is necessary to switch to the Bionic image on Travis? What is the oldest Ubuntu version officially supported by TRIQS 3.0?

**Comment by krivenko** [comment]
This [issue](https://github.com/TRIQS/triqs/pull/779#issuecomment-662873651) is apparently making a return in TRIQS 3.0 (see Travis clang compilation log)...

File: c++/g2.cpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ namespace pomerol2triqs {
       o     = -o;
     }
 
-    std::complex<double> res = (sqrt(2 * l + 1) / sqrt(o)) * std::pow(1_j, o + l) * boost::math::cyl_bessel_j(l + 0.5, o * pi / 2);
+    std::complex<double> res = (sqrt(2 * l + 1) / sqrt(o)) * std::pow(1i, o + l) * boost::math::cyl_bessel_j(l + 0.5, o * pi / 2);
```
Comment (krivenko) [inline c++/g2.cpp]: g++ complains:
```
/home/travis/build/krivenko/pomerol2triqs/c++/g2.cpp: In function ‘std::complex<double> pomerol2triqs::t_bar(int, int)’:

/home/travis/build/krivenko/pomerol2triqs/c++/g2.cpp:44:80: error: no matching function for call to ‘pow(__complex__ int, int)’

     std::complex<double> res = (sqrt(2 * l + 1) / sqrt(o)) * std::pow(1i, o + l) * boost::math::cyl_bessel_j(l + 0.5, o * pi / 2);

                                                                                ^
```

I guess you have to put `using namespace std::complex_literals;` somewhere.

Comment (Wentzell) [reply c++/g2.cpp]: The `using namespace` directive is already done in the TRIQS header that previously defined the operator `_j`.
This seems to rather be an issue with g++-7. This compiles just fine for clang and g++-8 and up, see [here](https://godbolt.org/z/rqP43G). Using `std::pow(std::complex<double>{1i}, o+l)` seems to work around this.
Do you want me to put this?

**Comment by Wentzell** [comment]
> Thank you so much, Nils!
> 
> I must try to build and run this on my machine before I merge.
> 
> The only question I have right now is whether it is necessary to switch to the Bionic image on Travis? What is the oldest Ubuntu version officially supported by TRIQS 3.0?

We only officially support Ubuntu 18.04 and 20.04. That is also why I changed the image in travis.

**Comment by Wentzell** [comment]
> This [issue](https://github.com/TRIQS/triqs/pull/779#issuecomment-662873651) is apparently making a return in TRIQS 3.0 (see Travis clang compilation log)...

Yeah it seems to be the same problem. I can replay your workaround also on the cpp2py master branch.

File: c++/g2.cpp (hunk 1 above)
Comment (krivenko) [reply c++/g2.cpp]: Since we want to target Ubuntu Bionic, which ships GCC 7, there should be a workaround.

**Comment by Wentzell** [comment]
> > This [issue](https://github.com/TRIQS/triqs/pull/779#issuecomment-662873651) is apparently making a return in TRIQS 3.0 (see Travis clang compilation log)...
> 
> Yeah it seems to be the same problem. I can replay your workaround also on the cpp2py master branch.

This should now be addressed in https://github.com/TRIQS/cpp2py/commit/98bd1b2d5e91d7ea4936f4d33c199a83c79671d1

**Comment by krivenko** [comment]
Thank you once again, Nils!
Everything looks fine now.

**Pull request merged**